### ISSUE TITLE
Create checkbox for tables and graphs to have different widths on Article pages

### ISSUE THREAD

**Issue by nancyt1** [mozilla/foundation.mozilla.org#10163]
We've been getting a few requests to have simple tables, advanced tables, and datawrapper graphs fit the width of the content. Currently they break out the bounds of the content and it would be great to have a checkbox like the one in the image block to display the graphs at wide or narrow widths. By default, it should align with the content. 

AC
- As an editor, I'm able to check a box to have my table wider than the content.
- As a reader, I'm able to see tables the same width as the content if "wide" box is unchecked.

**Comment by tbrlpld** [comment]
Do we ever want them to overflow the content width?

**Comment by nancyt1** [comment]
I don't think so. Can you think of any cases where that might happen? @tbrlpld

**Comment by tbrlpld** [comment]
@nancyt1 I does happen in the example you are showing above. Sorry, maybe I should have been more specific about my question. The question is: does this need to be an option or should we make sure that it always has the same width as the content?

**Comment by nancyt1** [comment]
@tbrlpld Yes, it should be an option. I was thinking we have a "wide" option like the screenshot above and by default it will have the same width as the content.

**Comment by tbrlpld** [comment]
Sorry, that was the question: Why do we want it to be wide? (that is what I mean with overflow... which I realize now has been poor wording)

**Comment by nancyt1** [comment]
I would like the keep the width that we have currently and just label it as "wide". We've just been getting requests to make the tables narrower to align with the content but the wide option is good for tables that have a lot of columns or graphs that need more room for details. Sorry does that make sense or would u want to hop on a huddle?

**Comment by tbrlpld** [comment]
@nancyt1 Thanks that makes sense. I guess it would have been a lot easier if I would have asked the initial question more straight clearly 😅

**Comment by tbrlpld** [comment]
@nancyt1 I assume the checkbox would be ok to have an effect on other pages where the block types exist too?

**Comment by nancyt1** [comment]
Yeah, I think that would be fine!

**Comment by tbrlpld** [comment]
@nancyt1 Considering that the current layout is the "wide" one, I assume the check box should default to "checked" (True).

**Comment by nancyt1** [comment]
That works!

**Comment by kristinashu** [comment]
@mtdenton @cdanfon any chance this could get into the upcoming sprint? There's another report launching around March 20 and it would be nice to have this feature.

**Comment by danielfmiranda** [comment]
Hi Everyone! 

After taking a look at this, I found that while we are able to add the "wide" field to the `AdvancedTableBlock` easily, adding the option to `DataWrapperBlock` and `TableBlock` are a bigger lift and something I probably will not have time for this sprint.

This is because the `AdvancedTableBlock` is built from a `StructBlock`, which Wagtail made to be customizable with any field of our choice. 

The `DatawrapperBlock` and `TableBlock` however, are extensions of Wagtail's `EmbedBlock` and `TableBlock`. Both predefined blocks without the option to add any extra fields.

In order to add the wide option to these blocks, we would need to redefine them as struct blocks, add the fields, and create a custom migration to make sure that the data for these blocks is not lost with the update.

Definitely doable, just a bigger lift then I have the bandwidth for this sprint.


I have a PR ready for the advanced table block: 
- [https://github.com/MozillaFoundation/foundation.mozilla.org/pull/10333](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/10333)

And have created two tickets for the other blocks:

- [https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10332](https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10332)
- [https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10331](https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10331)

**Comment by kristinashu** [comment]
Thank you Daniel for looking into this and summarizing it!

**Comment by danielfmiranda** [comment]
Hi @kristinashu! 

This ticket has been merged into staging and is now ready for testing. 

If you have time can you please take a look and let me know if everything is working as expected? 

Thanks!

**Comment by kristinashu** [comment]
Thank you @danielfmiranda! Maybe it's just that I haven't had my coffee yet but I don't see "wide" as an option in  Advanced tables on Article pages on staging. Is this where I should be seeing it?

<img width="1509" alt="image" src="https://user-images.githubusercontent.com/1068783/225657893-3726a524-bfe7-47dc-b9b7-0ec3a58d0810.png">
